src/main/webapp/app/quiz/edit/short-answer-question/edit-short-answer-question.component.ts

```diff
@@ -568,4 +584,115 @@ export class EditShortAnswerQuestionComponent implements OnInit, OnChanges, Afte
        this.questionEditor.getEditor().setValue(this.generateMarkdown());
        this.questionEditor.getEditor().clearSelection();
     }
+
+    /**
+     * For Re-evaluate
+     */
+
+    /**
+     * @function moveUp
+     * @desc Move this question one position up
+     */
+    moveUp() {
+        this.questionMoveUp.emit();
+    }
+
+    /**
+     * @function moveDown
+     * @desc Move this question one position down
+     */
+    moveDown() {
+        this.questionMoveDown.emit();
+    }
+
+    /**
+     * @function
+     * @desc Resets the question title
+     */
+    resetQuestionTitle() {
+        this.question.title = this.backupQuestion.title;
+    }
+
+    /**
+     * @function resetQuestionText
+     * @desc Resets the question text
+     */
+    resetQuestionText() {
+        this.question.text = this.backupQuestion.text;
+        this.question.spots = JSON.parse(JSON.stringify(this.backupQuestion.spots));
+        this.textParts = this.question.text.split(/\n+/g).map(t => t.split(/\s+(?![^[]]*])/g));
+        this.question.explanation = this.backupQuestion.explanation;
+        this.question.hint = this.backupQuestion.hint;
+    }
+
+    /**
+     * @function resetQuestion
+     * @desc Resets the whole question
+     */
+    resetQuestion() {
+        this.resetQuestionTitle();
+        this.question.invalid = this.backupQuestion.invalid;
+        this.question.randomizeOrder = this.backupQuestion.randomizeOrder;
+        this.question.scoringType = this.backupQuestion.scoringType;
+        this.question.solutions = JSON.parse(JSON.stringify(this.backupQuestion.solutions));
+        this.question.correctMappings = JSON.parse(JSON.stringify(this.backupQuestion.correctMappings));
+        this.question.spots = JSON.parse(JSON.stringify(this.backupQuestion.spots));
+        this.resetQuestionText();
+    }
+
+    /**
+     * @function resetSpot
+     * @desc Resets the spot
+     * @param spot {spot} the spot, which will be reset
+     */
+    resetSpot(spot: ShortAnswerSpot): void {
+        // Find matching spot in backupQuestion
+        const backupSpot = this.backupQuestion.spots.find(currentSpot => currentSpot.id === spot.id);
+        // Find current index of our spot
+        const spotIndex = this.question.spots.indexOf(spot);
+        // Remove current spot at given index and insert the backup at the same position
+        this.question.spots.splice(spotIndex, 1);
+        this.question.spots.splice(spotIndex, 0, backupSpot);
+    }
+
+    /**
+     * @function deleteSpot
+     * @desc Delete the given spot
+     * @param spotToDelete {object} the spot to delete
+     */
+    deleteSpot(spotToDelete: ShortAnswerSpot): void {
+        this.question.spots = this.question.spots.filter(spot => spot !== spotToDelete);
+        this.deleteMappingsForSpot(spotToDelete);
+
+        this.textParts = this.question.text.split(/\n+/g).map(t => t.split(/\s+(?![^[]]*])/g));
```

Comment:
use constants for regex expressions and make sure to comment them!

Reply:
adapted see comment above.